Library/Homebrew/utils/pypi.rb

```diff
@@ -318,10 +315,17 @@ def self.update_python_resources!(formula, version: nil, package_name: nil, extr
       if name.blank?
         odie "Unable to resolve some dependencies. Please update the resources for \"#{formula.name}\" manually."
       elsif url.blank? || checksum.blank?
-        odie <<~EOS
-          Unable to find the URL and/or sha256 for the "#{name}" resource.
-          Please update the resources for "#{formula.name}" manually.
-        EOS
+        existing_resource = formula.resource(name)
+        if existing_resource.present? && existing_resource.version == package.version
+          non_pypi_resources.delete existing_resource
```

Comment:
TODO:
- [ ] Try to piggy back on some formula bump logic to allow replacing version in URL.
- [ ] `non_pypi_resources.delete` on situation where a non-PyPI resource is now sdist on PyPI